test/core/event_engine/posix/posix_event_engine_connect_test.cc

```diff
@@ -0,0 +1,240 @@
+#include <errno.h>
+#include <fcntl.h>
+#include <poll.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <algorithm>
+#include <chrono>
+#include <cstring>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "absl/memory/memory.h"
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/str_cat.h"
+#include "gtest/gtest.h"
+
+#include <grpc/event_engine/event_engine.h>
+#include <grpc/grpc.h>
+#include <grpc/support/log.h>
+
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/event_engine/channel_args_endpoint_config.h"
+#include "src/core/lib/event_engine/posix_engine/posix_engine.h"
+#include "src/core/lib/event_engine/posix_engine/tcp_socket_utils.h"
+#include "src/core/lib/experiments/experiments.h"
+#include "src/core/lib/gprpp/notification.h"
+#include "src/core/lib/resource_quota/memory_quota.h"
+#include "src/core/lib/resource_quota/resource_quota.h"
+#include "test/core/event_engine/test_suite/event_engine_test_utils.h"
+#include "test/core/util/port.h"
+#include "test/core/util/test_config.h"
+
+namespace grpc_event_engine {
+namespace posix_engine {
+
+using ::grpc_event_engine::experimental::ChannelArgsEndpointConfig;
+using ::grpc_event_engine::experimental::EventEngine;
+using ::grpc_event_engine::experimental::PosixEventEngine;
+using ::grpc_event_engine::experimental::URIToResolvedAddress;
+using ::grpc_event_engine::experimental::WaitForSingleOwner;
+using namespace std::chrono_literals;
+
+namespace {
+
+// Creates a server socket listening for one connection on a specific port. It
+// then creates another client socket connected to the server socket. This fills
+// up the kernel listen queue on the server socket. Any subsequent attempts to
+// connect to the server socket will be pending indefinitely. This can be used
+// to test Connection timeouts and cancellation attempts.
+absl::StatusOr<std::vector<int>> CreateConnectedSockets(
+    EventEngine::ResolvedAddress resolved_addr) {
+  int server_socket;
+  int opt = -1;
+  int client_socket;
+  int one = 1;
+  int flags;
+  std::vector<int> ret_sockets;
+  // Creating a new socket file descriptor.
+  if ((server_socket = socket(AF_INET6, SOCK_STREAM, 0)) <= 0) {
+    return absl::UnknownError(
+        absl::StrCat("Error creating socket: ", std::strerror(errno)));
+  }
+  // MacOS biulds fail if SO_REUSEADDR and SO_REUSEPORT are set in the same
+  // setsockopt syscall. So they are set separately one after the other.
+  if (setsockopt(server_socket, SOL_SOCKET, SO_REUSEADDR, &opt, sizeof(opt))) {
+    return absl::UnknownError(
+        absl::StrCat("Error setsockopt(SO_REUSEADDR): ", std::strerror(errno)));
+  }
+  if (setsockopt(server_socket, SOL_SOCKET, SO_REUSEPORT, &opt, sizeof(opt))) {
+    return absl::UnknownError(
+        absl::StrCat("Error setsockopt(SO_REUSEPORT): ", std::strerror(errno)));
+  }
+
+  // Forcefully bind the new socket.
```

Comment:
Out of curiosity, why is this considered forecful?

Reply:
Not apt wording. I changed the comment.